src/Caching/SimpleFileCache.php

```diff
@@ -0,0 +1,159 @@
+        }
+
+        return self::writeToFile(
+            $this->directory,
+            $key,
+            '<?php return ' . var_export(['data' => $value], true) . ';'
+        );
+    }
+
+    private static function writeToFile(string $directory, string $filename, string $content): bool
+    {
+        if (! self::createDirectoryIfNeeded($directory) || ! is_writable($directory)) {
+            return false;
+        }
+
+        set_error_handler(self::$emptyErrorHandler);
+
+        $tmpFile = tempnam($directory, 'swap');
+        assert(is_string($tmpFile));
+
+        chmod($tmpFile, self::FILE_PERMISSIONS);
+
+        if (file_put_contents($tmpFile, $content) !== false) {
+            chmod($tmpFile, self::FILE_PERMISSIONS);
+
+            if (rename($tmpFile, $directory . $filename)) {
+                restore_error_handler();
+
+                return true;
+            }
+
+            unlink($tmpFile);
+        }
+
+        restore_error_handler();
+
+        return false;
+    }
+
+    private static function createDirectoryIfNeeded(string $directory): bool
+    {
+        if (is_dir($directory)) {
+            return true;
+        }
+
+        set_error_handler(self::$emptyErrorHandler);
+        $created = mkdir($directory, self::DIRECTORY_PERMISSIONS, true);
+        restore_error_handler();
+
+        return $created !== false || is_dir($directory);
+    }
+
+    /** @inheritDoc */
+    public function delete($key)
+    {
+        throw new LogicException(__METHOD__ . ' is not implemented. Use a full PSR-16 driver if you need support for that');
+    }
+
+    public function clear(): bool
+    {
+        throw new LogicException(__METHOD__ . ' is not implemented. Use a full PSR-16 driver if you need support for that');
+    }
+
+    /** @inheritDoc */
+    public function getMultiple($keys, $default = null): iterable
+    {
+        throw new LogicException(__METHOD__ . ' is not implemented. Use a full PSR-16 driver if you need support for that');
+    }
+
+    /** @inheritDoc */
+    public function setMultiple($values, $ttl = null): bool
+    {
+        throw new LogicException(__METHOD__ . ' is not implemented. Use a full PSR-16 driver if you need support for that');
+    }
+
+    /** @inheritDoc */
+    public function deleteMultiple($keys)
+    {
+        throw new LogicException(__METHOD__ . ' is not implemented. Use a full PSR-16 driver if you need support for that');
+    }
+
+    /** @inheritDoc */
+    public function has($key): bool
+    {
+        throw new LogicException(__METHOD__ . ' is not implemented. Use a full PSR-16 driver if you need support for that');
+    }
+
+    /** @return array<string, mixed>|null */
+    private static function readFileContents(string $filename): ?array
+    {
+        // note: error suppression is still faster than `file_exists`, `is_file` and `is_readable`
```

Comment:
will this supress errors like permission problems?  how material is the difference?  the readability of using the appropriate calls and the surety that we'll get errors reported seems pretty valuable to me.

Reply:
As #230 says, not being able to read the file or not having an array being provided should result in the same thing: cache should be warmed up.

We're talking about 3 function calls vs letting the `include` fail.